tests/testreactive.py

```diff
@@ -551,6 +576,24 @@ def gen():
             break
     assert rxgen.rx.value == 2
 
+async def test_reactive_gen_pipe():
+    def gen(val):
+        yield val+1
+        time.sleep(0.05)
+        yield val+2
+
+    rxv = rx(0)
+    rxgen = rxv.rx.pipe(gen)
+    rxgen.rx.watch()
+    assert rxgen.rx.value is param.Undefined
+    await asyncio.sleep(0.04)
+    assert rxgen.rx.value == 1
+    await asyncio.sleep(0.1)
+    assert rxgen.rx.value == 2
+    rxv.rx.value = 2
+    await asyncio.sleep(0.03)
+    assert rxgen.rx.value == 3
```

Comment:
I'm not entirely sure why the first one is 0.04, and the last one is 0.03. Is that just because it works?

Reply:
Solid question, really we need a `wait_until` utility like we have in Panel.

Reply:
Steal it